File: codingTest/baekjoon/javascript/10828.js
let inp = require("fs").readFileSync("/dev/stdin").toString().split("\n");

let stack = [];
let result = "";
for (let i = 1; i < inp.length; i++) {
  let arr = inp[i].split(" ");
  let len = stack.length;
  if (arr[0] == "push") stack.push(arr[1]);
  else if (arr[0] == "pop") result += (len == 0 ? -1 : stack.pop()) + "\n";
  else if (arr[0] == "size") result += len + "\n";
  else if (arr[0] == "empty") result += (len == 0 ? 1 : 0) + "\n";
  else if (arr[0] == "top") result += (len == 0 ? -1 : stack[len - 1]) + "\n";
}

console.log(result);

File: codingTest/baekjoon/python/10828.py
import sys

stack = []
for _ in range(int(sys.stdin.readline())):
    arr = sys.stdin.readline().split()
    if len(arr) == 2:
        stack.append(arr[1])
    else:
        if arr[0] == "pop":
            print(-1 if len(stack) == 0 else stack.pop())
        elif arr[0] == "size":
            print(len(stack))
        elif arr[0] == "empty":
            print(1 if len(stack) == 0 else 0)
        elif arr[0] == "top":
            print(-1 if len(stack) == 0 else stack[len(stack) - 1])
